### PR TITLE
add new test cases

### PR DESCRIPTION
**Pull request by sumukhmarathe**

### CODE REVIEW

File: src/core/fallcurve/internal/vt_fc_signature_compute.c [hunk 1]
```diff
@@ -95,12 +97,12 @@ VT_UINT fc_signature_compute(
...
     VT_INT index_37 = fc_signature_calculate_37index(raw_signature, sample_length);
     // fingerprint_length of this new fingerprint
-    if (index_37 > 0)
+    if (index_37 < sample_length)
     {
         sample_length = index_37 + 1;
     }
     // Calculate FallTime
-    falltime_computed = (VT_ULONG)sample_length * sampling_interval_us;
+    falltime_computed = (VT_ULONG)index_37 * sampling_interval_us;
```
Comment (ryanwinter) [inline src/core/fallcurve/internal/vt_fc_signature_compute.c:105]: is this a bug fix? Seems to be a change in logic.

File: src/core/fallcurve/internal/vt_fc_signature_compute.c [hunk 2]
```diff
@@ -33,7 +33,9 @@ static VT_INT fc_signature_calculate_37index(VT_UINT* raw_signature, VT_UINT sam
         }
     }
 
-    return -1;
+    return abs_custom(
```
Comment (ryanwinter) [inline src/core/fallcurve/internal/vt_fc_signature_compute.c:36]: This logic is confusing to me. I get before returning -1 meaning it didnt find a match, but what does it mean to return this value now?

File: src/core/fallcurve/internal/vt_fc_signature_collection_settings.c [hunk 3]
```diff
@@ -100,8 +99,7 @@ VT_UINT fc_signature_compute_collection_settings(
         return VT_ERROR;
     }
 
-    falltime_datapoints   = round((float)falltime_calib_test / (float)*sampling_interval_us);
-    *sampling_interval_us = round(((float)(*sampling_interval_us) * (float)falltime_datapoints) / (float)VT_FC_SAMPLE_LENGTH);
+    *sampling_interval_us = round((float)falltime_calib_test / (float)VT_FC_SAMPLE_LENGTH);
```
Comment (ryanwinter) [inline src/core/fallcurve/internal/vt_fc_signature_collection_settings.c:102]: This looks like a simplification of logic, is that right?

**Review by ryanwinter — CHANGES_REQUESTED** [review]
This appears to be more than just unit test changes, so we need to be careful about how we manage those additional changes. In the future, this would be better as two PR's.

File: src/core/fallcurve/internal/vt_fc_signature_collection_settings.c (hunk 3 above)
Comment (sumukhmarathe) [reply src/core/fallcurve/internal/vt_fc_signature_collection_settings.c:102]: yep simplification of logic, that's correct

File: src/core/fallcurve/internal/vt_fc_signature_compute.c (hunk 1 above)
Comment (sumukhmarathe) [reply src/core/fallcurve/internal/vt_fc_signature_compute.c:105]: Yes this is a change in logic for bug fix, explained in the reply to the next comment. I realised that this is not the right way to do it in terms of commits and PRs after I pushed it

File: src/core/fallcurve/internal/vt_fc_signature_compute.c (hunk 2 above)
Comment (sumukhmarathe) [reply src/core/fallcurve/internal/vt_fc_signature_compute.c:36]: So the fc_signature_calculate_37index function was returning -1 if the voltage response doesn't reach 37 % of the starting value by the time all SAMPLE_LENGTH data points are collected. When line 105 didn't use this returned value (index37), if -1 was returned by this function it entailed that legal and valid but SLOWER fallcurves which would reach 37% of starting value AFTER SAMPLE_LENGTH, would give same falltime as that of a fallcurve which reached 37% AT SAMPLE_LENGTH. Due to this bug, slower but legal/valid fallcurves were being successfully matched with fallcurves with high rate of voltage drop  as their falltimes were identical

**Review by ryanwinter — APPROVED** [review]

**Pull request merged**